### PR TITLE
Component-feature

### PR DESCRIPTION
**Pull request by gilberthappi**
Understand the concept of state management.
Use store, actions and reducers in React.
Connect React and Redux.

### CODE REVIEW

File: src/components/BookIndividual.jsx [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
+  };
+
+  return (
+    <div className="info">
+      <h2>{book.title}</h2>
+      <p>
+        Author:
+        {book.author}
+      </p>
+      <button type="button" className="btn" onClick={handleDelete}>Delete</button>
```
Comment (Johnadibe) [inline src/components/BookIndividual.jsx]: > - [ ] According to the project requirements as stated [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/use_redux_in_react_components_v2.md#:~:text=Add%20a%20%3CButton,by%20id), you should add a <Button> component and not button jsx that includes an event handler that removes a book from the books array.👍
> 
> ```
> const Button = ({ chilldren, onClick }) => {
>   return (
>     <button onClick={onClick}>{children}</button>
>   );
> };
> ```
> 
> In the BookForm component:
> 
> ```
> const BookForm = () => {
>  // Your states and handleSubmit Logic
> 
>   return (
>     <form>
>       <h3>Add a new book</h3>
>       <input
>         type="text"
>         placeholder="Book Title"
>         value={title}
>         onChange={(e) => setTitle(e.target.value)}
>       />
>       <input
>         type="text"
>         placeholder="Author"
>         value={author}
>         onChange={(e) => setAuthor(e.target.value)}
>       />
>       <Button onClick={handleSubmit}>Add a Book</Button>
>     </form>
>   );
> };
> ```
> 
> ```
> const BookIndividual = ({ book, handleDelete}) => {
>   // Your Logic
> 
>   return (
>     <div className="info">
>       <h2>{book.title}</h2>
>       <p>
>         Author:
>         {book.author}
>       </p>
>       <Button onClick={handleDelete}>Delete</Button>
>     </div>
>   );
> };
> ```
> 
> You can then write the function to handleDelete in your bookList component, (where you would render the BookIndividual component) and pass the handleDelete as a prop in the BookIndividual component.👍

**Review by Johnadibe — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️
### Hi @gilberthappi👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

mario_ladders

## Highlights
✅ Wrapped <App> with the <Provider> component from react-redux✔
✅ Passed configured store into the <Provider>✔
✅ Created an initialState variable for your booksSlice✔️
✅ Displayed books, received from the slice, in a reusable component✔
✅ Dispatched actions using useDispatch✔

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookIndividual.jsx (hunk 1 above)
Comment (Johnadibe) [reply src/components/BookIndividual.jsx]: > - [ ] Hello @gilberthappi, In line with the suggestions made by the previous reviewer, please create a <Button> component and apply it for both adding and removing functionality within the Bookstore application. You can employ the provided code examples as a reference. This approach aligns you with the project's requirements.👍

File: src/components/BookIndividual.jsx [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
+    <div className="info">
+      <h2>{book.title}</h2>
+      <p>
+        Author:
+        {' '}
+        {book.author}
+      </p>
+      <button type="button" className="btn" onClick={handleDelete}>
+        Delete
+      </button>
```
Comment (Johnadibe) [inline src/components/BookIndividual.jsx:23]: > - [ ] Hello @gilberthappi, In line with the suggestions made by the previous reviewer, please create a <Button> component and apply it for both adding and removing functionality within the Bookstore application. You can employ the provided code examples as a reference. This approach aligns you with the project's requirements.👍

**Review by Johnadibe — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️
### Hi @gilberthappi🙋‍♂️, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights
- No linter errors✔
- Your project is documented professionally✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookIndividual.jsx [hunk 3]
```diff
@@ -1,22 +1,36 @@
...
-    <button type="button" className="btn" onClick={() => onDelete(book.id)}>Delete</button>
-  </div>
-);
+const BookIndividual = ({ book, onDelete }) => {
+  const dispatch = useDispatch();
+
+  const handleDelete = () => {
+    dispatch(removeBook(book.item_id));
+    onDelete(book.item_id);
+  };
```
Comment (thecodechaser) [inline src/components/BookIndividual.jsx]: - [ ] Delete button is working fine for default books available in store but if I add books from the book form and try to delete any books It's deleting two books that are added from the forms. Please make sure to fix this, (To reproduce this, Try to add more than one book from the form and try to delete any of the books).

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @gilberthappi👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- Good job with redux code. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi @gilberthappi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Addressed changes requested



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**